src/plugins/discover/public/application/main/components/layout/discover_histogram_layout.tsx

```diff
@@ -47,6 +48,8 @@ export const DiscoverHistogramLayout = ({
   if (!searchSessionId && !isPlainRecord) {
     return null;
   }
+  const textBasedQueryColumns = dataState.data$.documents$.getValue().textBasedQueryColumns;
```

Comment:
```suggestion
  const textBasedQueryColumns = useObservable(
    dataState.data$.documents$,
    dataState.data$.documents$.getValue()
  ).textBasedQueryColumns;
```
Can we use `useObservable` here to make sure it's reactive? It doesn't seem to matter currently but better to do it now in case of future changes.

Reply:
Done https://github.com/elastic/kibana/pull/165557/commits/15b234f25bdbcbb03292cf9331ff0468e0aa4351